documentation/design-docs/dotnet-monitor.yaml

```
@@ -0,0 +1,473 @@
+      tags:
+      - "metrics"
+      operationId: "getdefaultmetrics"
+      produces:
+      - "text/plain; version=0.0.4"
+      - "application/json"
+      responses:
+        200:
+          description: "Success"
+          schema:
+           $ref: "#/definitions/Metrics" 
+        400:
+          description: "Bad request"
+          schema:
+            $ref: "#/definitions/Error"
+  /metrics/{uid}:
+    get:
+      tags:
+      - "metrics"
+      operationId: "getmetrics"
+      produces:
+      - "text/plain; version=0.0.4"
+      - "application/json"
+      parameters:
+      - name: "uid"
+        in: "path"
+        description: "Process ID of the process"
+        required: true
+        type: "string"
+      responses:
+        200:
+          description: "Success"
+          schema:
+           $ref: "#/definitions/Metrics" 
+        400:
+          description: "Bad request"
+          schema:
+            $ref: "#/definitions/Error"
+    post:
+      tags:
+      - "metrics"
+      operationId: "postmetrics"
+      produces:
+      - "text/plain; version=0.0.4"
+      parameters:
+      - name: "uid"
+        in: "path"
+        description: "Process ID of the process"
+        required: true
+        type: "string"
+      - name: "configuration"
+        in: "body"
+        schema:
+          type: "array"
+          items:
+            $ref: "#/definitions/EventPipeProvider"
+      responses:
+        200:
+          description: "Success"
+          schema:
+           $ref: "#/definitions/Metrics" 
+        400:
+          description: "Bad request"
+          schema:
+            $ref: "#/definitions/Error"
+
+# Descriptions of common components
+definitions:
+  # Prometheus line exposition format
+  Metrics:
+    type: string
+    example: "http_requests_total{method='post',code='200'} 1027 1395066363000\nhttp_requests_total{method='post',code='400'}    3 1395066363000"
+  Process:
+    type: object
+    properties:
+      uid:
+        type: string
+      name:
+        type: string
+      commandLine:
+        type: string
+    required:
+      - uid
+  LogFilter:
+    type: object
+    properties:
+      provider:
+        type: string
+      logLevel:
+        type: string
+        enum:
+        - "Trace"
+        - "Debug"
+        - "Information"
+        - "Warning"
+        - "Error"
+        - "Critical"
+  # RFC 7807 Problem details
+  Error:
+    type: object
+    properties:
+      code:
+        type: string
+      title:
+        type: string
+      status:
+        type: integer
+      detail:
+        type: string
+      instance:
+        type: string
+    required:
+      - code
+      - title
+      - status
+      - detail
+      - instance
+  EventPipeConfiguration:
+    type: object
+    properties:
+      providers:
+        type: array
+        items:
+          $ref: "#/definitions/EventPipeProvider"
+      requestRundown:
+        type: boolean
+        default: true
+      circularBufferInMB:
```

```suggestion
      bufferSizeInMB:
```
(The buffer isn't actually circular, when it overflows the new data is dropped rather than the old data)